src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -112,11 +115,25 @@ public override void Initialize(AnalysisContext analysisContext)
                         continue;
                     }
 
-                    CheckArgument(owningSymbol, creation, argument.Parameter, value, context);
+                    diagnostic = CheckArgument(owningSymbol, creation, argument.Parameter, value, context);
```

Comment:
Nit: can we consider `CheckArgument` itself to report the diagnostic instead of passing it back to the caller? It can be changed to return a bool, and we can break out from this loop when it returns false.

Reply:
The position of `parameterName` and `message` are different accrosss ArgumentException descendants, we want to diagnostic for `message` suppressed `parameterName` diagnostic, For ArgumentException descendants `message` is 2nd parameter and `parameterName` diagnostic would win if we report that diagnostic and break. Also we should trigger fixer only for `message` diagnostic